src/analysis/graphspace.py

```diff
@@ -55,12 +59,27 @@ def get_gs_graph(graph_file:str,graph_name:str,directed=False) -> GSGraph:
 			G.add_edge_style(u,v,width=2,color='#281D6A')
 	return G
 
-## TODO this is a duplicated function in summary.py.
-## Pull this and others into a utils.py function.
-def load_graph(path: str,directed=False) -> nx.Graph:
-	if not directed:
-		G = nx.read_edgelist(path,data=(('rank',float),))
+
+def load_graph(path: str) -> nx.Graph:
+	G = nx.Graph()
+	directed = False
+
+	try:
+		pathways = pd.read_csv(path, sep="\t", header=None)
+	except pd.errors.EmptyDataError:
+		print(f"The file {path} is empty.")
+		return G, directed
+	pathways.columns = ["Interactor1", "Interactor2", "Rank", "Direction"]
+	mask_u = pathways['Direction'] == 'U'
+	mask_d = pathways['Direction'] == 'D'
+
+	if mask_u.all():
+		G = nx.read_edgelist(path,data=(('rank',float), ('Direction',str)))
+		directed = False
+	elif mask_d.all():
+		G = nx.read_edgelist(path,data=(('rank',float),('Direction',str)), create_using=nx.DiGraph)
+		directed = True
 	else:
-		# note - self-edges are not allowed in DiGraphs.
-		G = nx.read_edgelist(path,data=(('rank',float),),create_using=nx.DiGraph)
-	return G
+		print("graphspace does not deal with mixed direction type graphs currently")
```

Comment:
```suggestion
		print("GraphSpace does not deal with mixed direction type graphs currently")
```
Do we need to print the filename or some identifier for the graph as part of this message saying we could not not visualize graph X?